### ISSUE TITLE
Better define source location expectations

### ISSUE THREAD

**Issue by nzakas** [estree/estree#80]
I'm noticing that there are some slight location differences between Esprima and Acorn.

Consider the following:

```js
function bar(a) {
    switch (a) {
        case 2:
            break;
        case 1:
            break;
        //no default
    }
}

```

In case it's not obvious, I have an intentional empty line at the end of that code (automatically inserted by my editor at the end of files).

In this case, Esprima says that the program range is `[0, 133]` while Acorn says the range is `[0, 134]`.

For locations, the start is the same but the end is different. Esprima says the program ends on `{ column: 1, line: 9 }` while Acorn says the program ends on `{ column: 0, line: 10 }`.

From these two differences, it seems that Acorn is including the trailing whitespace as part of the program while Esprima is not. I'm not sure which is correct, but it would be nice to have consistency. :)

**Comment by nzakas** [comment]
@RReverser any thoughts on this?

**Comment by RReverser** [comment]
@nzakas I agree these things should be standardized, and not necessarily related to CST (@mikesherov) but rather just to definition of what we consider part of node and what not. Somewhat related to #66 - if we would explicitly define which syntax string is represented by given node type, it would be clear which location part of source string it should cover.

**Comment by danez** [comment]
The same is true for newlines and comments at the beginning of a program, see https://github.com/babel/babel/issues/6681.
